### PR TITLE
Third party audit exceptions

### PR DESCRIPTION
**Pull request by alpar-t**
- fix up to date checks to ignore elasticsearch jars.
   - we were not scanning them but these still triggered a rebuild unnecessarily invalidating the cache 
- add tests to assert correct behavior and up to date checks.
    - we use a local repo with some dummy jars in it
- make the task less verbose with `-i` and include the output only on errors.

### CODE REVIEW

**Review by nik9000 — COMMENTED** [review]
Do you mean to commit the jars? It feels weird to commit them rather than build them. I don't like having stuff that we can't regenerate easily in the repo and having binary blobs in the repo can really bloat it. They aren't huge, but it is the principal of the thing!

File: buildSrc/src/main/java/org/elasticsearch/gradle/precommit/ThirdPartyAuditTask.java [hunk 1]
```diff
@@ -139,9 +130,25 @@ public void setExcludes(String... classes) {
         return Collections.unmodifiableSet(excludes);
     }
 
+    @InputFiles
+    @SkipWhenEmpty
+    public Set<File> getJarsToScan() {
+        Spec<Dependency> reallyThirdParty = dep -> dep.getGroup() == null || dep.getGroup().startsWith("org.elasticsearch") == false;
```
[the remark this answers is not in the document]
Comment (atorok) [reply buildSrc/src/main/java/org/elasticsearch/gradle/precommit/ThirdPartyAuditTask.java]: Good catch, i'we been meaning to go back and check what's up with the null  and forgot about it. Thanks !
I did some checks and In this context that's  not necessarily true, project deps will have groups. These are special dependencies mostly expressed with things like `compile files(...)` which have no way of having a group. I made changes to always scan those. They might not be third party but we have no way of knowing and are better off erring on scanning them.

**Comment by atorok** [comment]
@nik9000 I did check in the jars intentionally. I fully share your sentiment and would never do this outside of a test. I built these ad-hoc on the command line because they have a single `.class` file with something really simple like calling `new File` in the constructor and that's all the jar has. I felt like the extra work of setting up the test so these would be built and used in the test would make it harder to understand, and would be awkward if someone stumbles on the sources without relating them to the tests, especially since we'll also have to build a broken pom along the way to get the missing classes. I can't foresee a reason for which we'll ever want to regenerate these.
That being said, I'm fully open to suggestions and willing to get rid of the jars.

**Comment by nik9000** [comment]
I'm mostly worried that the jars will end up mysterious after a few years. If you build the jars on the fly then you automatically get documentation for how you built them.

**Comment by atorok** [comment]
@nik9000 I made changes in b29ca26 to build the jars as part of the test.

File: buildSrc/src/testKit/thirdPartyAudit/build.gradle [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    targetCompatibility = JavaVersion.VERSION_11
+    signatureFile = file('third-party-audit-empty.txt')
+}
+
+task absurd(type: ThirdPartyAuditTask) {
+    targetCompatibility = JavaVersion.VERSION_11
+    signatureFile = file('third-party-audit-absurd.txt')
+}
+
+// ---------------- SETUP ----------------
```
Comment (nik9000) [inline buildSrc/src/testKit/thirdPartyAudit/build.gradle]: Is this a leftover?

File: buildSrc/src/testKit/thirdPartyAudit/sample_jars/build.gradle [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    id 'java'
+}
+repositories {
+    mavenCentral()
+}
+dependencies {
+    compile 'org.apache.logging.log4j:log4j-core:2.11.1'
+}
+
+clean.enabled = false
```
Comment (nik9000) [inline buildSrc/src/testKit/thirdPartyAudit/sample_jars/build.gradle]: This seems weird. At least it is worth a comment about why we need it. Do we need it?

**Review by nik9000 — COMMENTED** [review]
I left two small questions.

**Review by nik9000 — APPROVED** [review]
LGTM

**Pull request merged**